protocol/xmrmaker/claim.go

```diff
@@ -135,41 +193,93 @@ func claimRelayer(
 		return ethcommon.Hash{}, err
 	}
 
-	abi, err := abi.JSON(strings.NewReader(contracts.SwapFactoryMetaData.ABI))
+	calldata, err := getClaimTxCalldata(relayerCommission, contractSwap, secret)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	feeValue, err := calculateRelayerCommission(contractSwap.Value, relayerCommission)
+	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	calldata, err := abi.Pack("claimRelayer", *contractSwap, secret, feeValue)
+	err = waitAndCheck(ctx, ec, txHash)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
+	return txHash, nil
+}
+
+func waitAndCheck(ctx context.Context, ec *ethclient.Client, txHash ethcommon.Hash) error {
+	waitDuration := time.Second
+	maxNotFound := 5
```

Comment:
You can declare these 2 as constants and add a 3rd constant for the total wait.  Here's what I came up with:
```
	const (
		checkInterval = time.Second // time between transaction polls
		maxWait       = time.Minute // max wait for first TX confirmation
		maxNotFound   = 5           // max failures where the TX is not even found in the mempool
	)
```

Reply:
done